packages/@aws-cdk/aws-cloudfront/lib/origin.ts

```diff
@@ -115,6 +168,25 @@ export abstract class Origin {
     return undefined;
   }
 
+  private renderCustomHeaders(): CfnDistribution.OriginCustomHeaderProperty[] | undefined {
+    if (!this.customHeaders || Object.entries(this.customHeaders).length === 0) { return undefined; }
+    return Object.entries(this.customHeaders).map(([headerName, headerValue]) => {
+      return { headerName, headerValue };
+    });
+  }
+
+  /**
+   * If the path is defined, it must start with a '/' and not end with a '/'.
+   * This method takes in the originPath, and returns it back (if undefined) or adds/removes the '/' as appropriate.
+   */
+  private validateOriginPath(originPath?: string): string | undefined {
+    if (originPath === undefined) { return undefined; }
```

Comment:
break if unresolved token

Reply:
What's the appropriate way to handle this? If the input is a token at this point, it can't be validated. So just take the token as-is, and skip validation, or check & throw an error?

```suggestion
    if (!Token.isUnresolved(originPath) && originPath === undefined) { return undefined; }
```

Reply:
Yap, don't validate if it's a token.